### ISSUE TITLE
AddOrUpdateAsync Throwing FabricObjectClosedException

### ISSUE THREAD

**Issue by Izzmo** [Azure/service-fabric-issues#404]
I have some code:

```csharp
            var markets = await PartnersMarketList.AddOrUpdateAsync(
               tx,
               partnerId,
               pId => new Dictionary<string, Market>
               {
                  { market.MarketCode, market }
               },
               (pId, currentPartnerMarketList) =>
               {
                  if (!currentPartnerMarketList.ContainsKey(market.MarketCode))
                     throw new RecordNotFoundException(nameof(market), market.MarketCode);

                  return new Dictionary<string, Market>(currentPartnerMarketList)
                  {
                     { market.MarketCode, market }
                  };
               });
```

Inside the update action, if I throw a custom exception, `RecordNotFoundException`, then this ends up throwing a `FabricObjectClosedException` which seems a bit odd.

As a workaround, I have it returning `null` right now and then catching that reference on the outside and then throwing my custom `RecordNotFoundException`.

So, my question is, is this correct functionality, or should it instead be rethrowing this exception up so I can correctly handle the custom exception outside.

**Comment by masnider** [comment]
Confirmed that this is a bug. We're tracking it internally now and will provide more information when we're shipping a fix. Thanks for reporting this!

**Comment by zuhairp** [comment]
Thanks again for reporting @Izzmo, the fix will be coming soon in our 6.2 release

**Comment by Izzmo** [comment]
Great! Thanks for following up @zuhairp

**Comment by masnider** [comment]
This fix shipped in 6.2. I'm going to close out this issue. Please let us know if you're still experiencing this and reopen the issue if so! Thanks again for reporting.
